src/lib/components/StoreUpdater.svelte

```diff
@@ -1,34 +1,29 @@
 <script lang="ts">
 	import { socket } from '$lib/stores/stateStore';
-	import { svgs, type Svg } from '$lib/stores/svgStore';
+	import { svgs } from '$lib/stores/svgStore';
 	import { onMount } from 'svelte';
+	import type { Svg } from '$lib/types';
+	import type { Edit, Remove } from '$lib/socketioInterface';
 
 	onMount(() => {
 		$socket.on('edit', handleEdit);
 		$socket.on('remove', handleRemove);
 	});
 
-	// TODO: Mads, optionals i interface.
-	function handleEdit(data: any) {
+	function handleEdit(data: Edit) {
 		// check if the command already exists
 		const commandIndex = $svgs.findIndex((svg: Svg) => svg.commandId === data.commandId);
 
 		// if it exists, modify it
-		// HACK: Maybe rewrite to loop through the keys of data instead
 		if (commandIndex !== -1) {
-			// only modify the property if it exists on the data object
-			if ('x' in data) {
-				$svgs[commandIndex].x = data.x;
-			}
-			if ('y' in data) {
-				$svgs[commandIndex].y = data.y;
-			}
-			if ('svg' in data) {
-				$svgs[commandIndex].svg = data.svg;
-			}
+			$svgs[commandIndex].offset = data.placement ?? $svgs[commandIndex].offset;
+			$svgs[commandIndex].svgString = data.svgString ?? $svgs[commandIndex].svgString;
 			return;
 		}
-		// else add it
+
+		// else add it, if all fields are present
+		if (!data.svgString || !data.placement) return;
```

Comment:
Efter min mening: Prøv at undgå at bruge ! foran her, da det ikke er intuitivt hvad det betyder for en ikke javascript-programmør